src/querystring.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{error::Error, Context};
+use http::StatusCode;
+use serde::Deserialize;
+
+/// An extension trait for `Context`, providing query string deserialization.
+pub trait ExtractQuery<'de> {
+    fn url_query<T: Deserialize<'de>>(&'de self) -> Result<T, Error>;
+}
+
+impl<'de, Data> ExtractQuery<'de> for Context<Data> {
+    fn url_query<T: Deserialize<'de>>(&'de self) -> Result<T, Error> {
```

Comment:
I think we could mark this function as `#[inline]` without a problem. Though we should probably do a full pass over Tide to inline functions where possible.

Reply:
I'm not entirely sure what that attribute does. Do you think it's something I should add now or wait for a later, crate-wide audit?

Reply:
No sooner had I asked than I found this. https://doc.rust-lang.org/nightly/reference/attributes/codegen.html#the-inline-attribute

It seems like a good idea to mark this as inline.  I'll try it when I get back to the computer.

Reply:
No sooner had I asked than I found this. https://doc.rust-lang.org/nightly/reference/attributes/codegen.html#the-inline-attribute

It seems like a good idea to mark this as inline.  I'll try it when I get back to the computer.

Reply:
`#[inline]` has been added.